ipfsapi/multipart.py

```diff
@@ -317,6 +316,56 @@ def body(self):
             yield chunk
 
 
+def glob_compile(pat):
+    """Translate a shell glob PATTERN to a regular expression.
+
+    This is almost entire based on `fnmatch.translate` source-code from the
+    python 3.5 standard-library.
+    """
+
+    i, n = 0, len(pat)
+    res = ''
+    while i < n:
+        c = pat[i]
+        i = i + 1
+        if c == '/' and len(pat) > (i + 2) and pat[i:(i + 3)] == '**/':
+            # Special-case for "any number of sub-directories" operator since
+            # may also expand to no entries:
+            #  Otherwise `a/**/b` would expand to `a[/].*[/]b` which wouldn't
+            #  match the immediate sub-directories of `a`, like `a/b`.
+            i = i + 3
+            res = res + '[/]([^/]*[/])*'
+        elif c == '*':
+            if len(pat) > i and pat[i] == '*':
```

Comment:
i love what you guys are doing, and would relicense to LGPL -- can you open a ticket?